aii-pxelinux/src/main/perl/pxelinux.pm

```diff
@@ -438,52 +493,59 @@ sub pxelink
 }
 
 # Prints the status of the node.
+# Display information for both PXELINUX and Grub2 variant.
+
 sub Status
 {
     my ($self, $cfg) = @_;
 
     my $t = $cfg->getElement (ETH)->getTree;
-    my $dir = $this_app->option (NBPDIR);
-    my $fqdn = get_fqdn($cfg);
-    my $boot = $this_app->option (LOCALBOOT);
-    my $rescue = link_filepath($cfg, RESCUE);
-    my $firmware = link_filepath($cfg, FIRMWARE);
-    my $livecd = link_filepath($cfg, LIVECD);
-    foreach my $s (values (%$t)) {
-        next unless $s->{ip};
-        my $ln = hexip ($s->{ip});
-        my $since = "unknown";
-        my $st;
-        if (-l "$dir/$ln") {
-            $since = ctime(lstat("$dir/$ln")->ctime());
-            my $name = readlink ("$dir/$ln");
-            my $name_path = "$dir/$name";
-            if (! -e $name_path) {
-                $st = "broken";
-            } elsif ($name =~ m{^(?:.*/)?$fqdn\.cfg$}) {
-                $st = "install";
-            } elsif ($name =~ m{^$boot$}) {
-                $st = "boot";
-            } elsif ($firmware && ($name_path =~ m{$firmware})) {
-                $st = "firmware";
-            } elsif ($livecd && ($name_path =~ m{$livecd})) {
-                $st = "livecd";
-            } elsif ($rescue && ($name_path =~ m{$rescue})) {
-                $st = "rescue";
+
+    foreach my $variant (PXE_VARIANT_PXELINUX, PXE_VARIANT_GRUB2) {
```

Comment:
As said inhttps://github.com/quattor/aii/pull/225#pullrequestreview-23865060, I didn't pay attention to the status so far, so this may need some improvement, you are right..., I wanted to get the configuration part working as a first step! I am not completely sure how to look at the configuration consistency between the 2 variants but I'll think about it (having an entry in both is not necessarily enough to have a consistent configuration but we could at least check that for a given ip, it points either to localboot or to the PXE config file).